compiler/ballerina-backend-llvm/src/main/ballerina/src/compiler_backend_llvm/llvm_gen.bal

```diff
@@ -169,3 +203,133 @@ function appendAllTo(any[] toArr, any[] fromArr) {
         i = i + 1;
     }
 }
+
+function getTaggedStructType(string typeName) returns llvm:LLVMTypeRef {
+    boolean hasType = structMap.hasKey(typeName);
+    if (!hasType) {
+        error typeNotInMapError = error("SimpleErrorType", message = "Type : " + typeName + " Not in structMap");
+        panic typeNotInMapError;
+    }
+    return <llvm:LLVMTypeRef>structMap[typeName];
+}
+
+function createUnion(bir:BUnionType bType) returns llvm:LLVMTypeRef {
+    string largestType = "null";
+    foreach var member in bType.members {
+        if (member is bir:BTypeInt) {
+            largestType = getLargerType(largestType, "int");
+            createTaggedInt();
```

Comment:
This seemingly looks like a random place to create the tagged struct type. Shall we rather move the "createTaggedInt" and "createTaggedBool" calls inside to "getTaggedStructType". So what it does is, it lazily create the types when someone looks up the tagged struct types using "getTaggedStructType". The idea is, now the creation of these struct types are bound to a certain flow in the code. This may not be apparent when this codebase grows, and we actually want to look up the tagged struct types without the "createUnion" call actually not being done. This may not happen practically, but we have to leave room for it to be possible. 

So basically, in any place we do "<llvm:LLVMTypeRef>structMap[largestType]", we should do "getTaggedStructType".

Reply:
I got the gist of it. I'll add the changes necessary.

Reply:
Done